library/ui/src/main/java/com/google/android/exoplayer2/ui/SubtitlePainter.java

```diff
@@ -312,7 +315,7 @@ private void setupBitmapLayout() {
     float anchorX = parentLeft + (parentWidth * cuePosition);
     float anchorY = parentTop + (parentHeight * cueLine);
     int width = Math.round(parentWidth * cueSize);
-    int height = Math.round(width * ((float) cueBitmap.getHeight() / cueBitmap.getWidth()));
+    int height = Math.round(width * ((float) cueBitmap.getHeight() / cueBitmap.getWidth()) / (((float) parentWidth / parentHeight) / cueSar));
```

Comment:
You can also get rid of the 1.7777 constant if you do this, and default to "natural height", which seems nicer.